webroot/src/components/Lists/Pagination/Pagination.ts

```diff
@@ -34,12 +34,16 @@ const createPaginationItem = (pageNum, currentPage) => ({
         InputSelect,
         LeftCaretIcon,
         RightCaretIcon
-    }
+    },
 })
 export default class Pagination extends mixins(MixinForm) {
-    @Prop({ required: true }) private pageChange!: (firstIndex: number, lastIndexExclusive: number) => any;
+    @Prop({ required: true }) private pageChange!:
+        (firstIndex: number, lastIndexExclusive: number, prevNext: number) => any;
+
     @Prop({ required: true }) private listSize!: number;
     @Prop({ required: true }) private paginationId!: string;
+    @Prop({ required: true }) pagingPrevKey!: string | null;
+    @Prop({ required: true }) pagingNextKey!: string | null;
     @Prop() private pageSizeConfig?: Array<{ value: number; name: string; isDefault?: boolean }>;
     @Prop() private ariaLabel?: string;
```

Comment:
I think this might address a few of my other questions... could we include a detailed file- or class- level comment describing the pagination model? A brief summary describing how you landed at this set of state, and explaining (as necessary) what each piece of state means.

If you think this better fits into a README, that works too.

Reply:
These are more props passed in from a parent to configure the pagination instance. I added some line comments for these. I can easily see how they could use some extra clarity 🙂  - especially with the less-conventional server paging in the API. I'm not 100% sure if we're done with the churn there - so this could still evolve some.

Reply:
Thank you!